satpy/tests/reader_tests/test_gerb_l2_hr_h5.py

```diff
@@ -0,0 +1,217 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Copyright (c) 2018 Satpy developers
+#
+# This file is part of satpy.
+#
+# satpy is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later
+# version.
+#
+# satpy is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# satpy.  If not, see <http://www.gnu.org/licenses/>.
+"""Unit tests for GERB L2 HR HDF5 reader."""
+
+import numpy as np
+import pytest
+import h5py
+from satpy import Scene
+
+FNAME = "G4_SEV4_L20_HR_SOL_TH_20190606_130000_V000.hdf"
+
+
+@pytest.fixture(scope="session")
+def gerb_l2_hr_h5_dummy_file(tmp_path_factory):
+    """Create a dummy HDF5 file for the GERB L2 HR product."""
+    filename = tmp_path_factory.mktemp("data") / FNAME
```

Comment:
The code quality checkers are a little upset with how big this function is. How much of this is used in the actual reading code? Is this a complete copy of the files structure? If it isn't all used you could remove it from the test. A lot of it also looks like it could go into a function (copy C_S1, set size, create variable, write, etc). Maybe even a for loop. Thoughts?

Reply:
Indeed, this is code generated by a routine that will duplicate some instructions. A lot actually. And very little of the file content is currently used in the code. Basically, the reader loads the most used data and makes sure that the geolocation is correct.